api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -723,8 +724,12 @@ public String getSecretQuestion(User user) throws APIException {
 	 */
 	@Override
 	@Transactional(readOnly = true)
-	public User getUserByEmail(String email) throws APIException {	
-		return dao.getUserByEmail(email);
+	public User getUserByEmail(String email) {
+		if (EmailValidator.getInstance().isValid(email)) {
```

Comment:
You're only supposed to check if the email is not null or blank, it's not this method's job to validate the email.